### ISSUE TITLE
[Feature Request] Yield Pool using coroutines and More Effective Coroutines

### ISSUE THREAD

**Issue by GilbertoBitt** [QFSW/MasterObjectPooler2#3]
how can i make a coroutine wait for the initialization of pool to avoid hiccup on the process loading or middle of the game?

**Comment by GilbertoBitt** [comment]
also do you plan to implement addressables?

**Comment by QFSW** [comment]
Initializing a pool is creating the objects which is strictly blocking in nature, but you could create only a single object per frame and wait on that

```cs
public IEnumerator PopulateCoroutine(ObjectPool pool, int count)
{
    for (int i = 0; i < count; i++)
    {
        pool.Populate(1, PopulateMethod.Add);
        yield return null;
    }
}
```

Addressables is an independent topic, plese make a new issue for it. In which way do you expect them to be implemented or used in MOP2?
